File: src/test/java/com/letgo/book/acceptance/CreateBookFeatureTest.java
package com.letgo.book.acceptance;

import org.junit.jupiter.api.Test;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import java.util.UUID;

import static org.junit.jupiter.api.Assertions.*;

final public class CreateBookFeatureTest extends TestCase {
    @Test
    public void itShouldCreateABook() {
        UUID id = UUID.randomUUID();
        ResponseEntity<String> postResponse = post("{'id': " + id + ", 'title': 'OlaKeAse'}");
        assertSame(HttpStatus.CREATED, postResponse.getStatusCode());
        assertFalse(postResponse.hasBody());

        weWaitForEventsToBeProcessed();

        ResponseEntity<String> getResponse = get("/book/" + id);
        assertEquals("OLAKEASE", getResponse.getBody());
    }
}

File: src/test/java/com/letgo/book/acceptance/CreateBookFeatureTest.kt
package com.letgo.book.acceptance

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertSame
import org.junit.jupiter.api.Test
import org.springframework.http.HttpStatus
import java.util.UUID

private class CreateBookFeatureTest : TestCase() {
    @Test
    fun `It should create a book`() {
        val id = UUID.randomUUID()

        post(
            body = """
                {
                    "id": "$id",
                    "title": "OlaKeAse"
                }
            """.trimIndent()
        ).run {
            assertSame(HttpStatus.CREATED, statusCode)
            assertFalse(hasBody())
        }

        weWaitForMessagesToBeProcessed()

        get(endpoint = "/book/$id").run {
            assertEquals("OLAKEASE", body)
        }
    }
}
